api/src/task/opportunities/import_opportunity_csvs.py

```diff
@@ -0,0 +1,72 @@
+import csv
+import logging
+from typing import cast
+
+import click
+
+import src.adapters.db as db
+import src.adapters.db.flask_db as flask_db
+import src.util.file_util as file_util
+from src.constants.schema import Schemas
+from src.task.task_blueprint import task_blueprint
+
+logger = logging.getLogger(__name__)
+
+FILES_TO_LOAD = [
+    "opportunity.csv",
+    "opportunity_summary.csv",
+    "opportunity_assistance_listing.csv",
+    "link_opportunity_summary_applicant_type.csv",
+    "link_opportunity_summary_funding_instrument.csv",
+    "link_opportunity_summary_funding_category.csv",
+]
+
+
+@task_blueprint.cli.command(
+    "import-opportunity-csvs",
+    help="Load several csv files to the opportunity tables",
+)
+@click.option("--input-folder", required=True, help="The directory to fetch the input files from")
+@flask_db.with_db_session()
+def import_opportunity_csvs(db_session: db.Session, input_folder: str) -> None:
+    with db_session.begin():
+        process(db_session, input_folder, Schemas.API)
+
+
+def process(db_session: db.Session, input_folder: str, schema: str) -> None:
+    for csv_file in FILES_TO_LOAD:
+        logger.info("Processing %s", csv_file)
+        table_name = csv_file.removesuffix(".csv")
+        csv_filepath = file_util.join(input_folder, csv_file)
+
+        load_csv_stream_to_table(db_session, table_name, csv_filepath, schema)
+
+
+def load_csv_stream_to_table(
+    db_session: db.Session, table_name: str, csv_filepath: str, schema: str
+) -> None:
+    # This is a bit hacky - I need all of the field names of the csv
+    # to write the COPY command, so open the file, read a single record
+    # so that we have the fieldnames, and then close the file
+    field_names: list[str] = []
+    with file_util.open_stream(csv_filepath) as csvfile:
+        reader = csv.DictReader(csvfile)
+
+        _ = reader.__next__
```

Comment:
Does this do anything? It's not being called, since it's missing `()`. Maybe DictReader gets the fieldnames without it.

Reply:
I think you're right. I had added it as I thought I remembered needing to read at least one line for dict reader to populate the field names, but removing it, it still works.

Looks like fieldnames literally does this in the property (although it looks like its done it for 16+ years, so no idea what I was remembering):
https://github.com/python/cpython/blob/main/Lib/csv.py#L160